### ISSUE TITLE
Raspberry Pi/Visual Basic - unable to compile data structure representing points of a polygon

### ISSUE THREAD

**Issue by zwolfpack** [mono/mono#18657]
Hello, 

I'm having issues in a VB app with a structure representing the points of a polygon. Both structures shown in the below test cases compile and work correctly under Windows Visual Studio, but fail to compile under vbnc.

Running mono under Linux (buster) on a Raspberry Pi:
```
pi@HS3-Pi:~/vb $ uname -a
Linux HS3-Pi 4.19.93-v7+ #1290 SMP Fri Jan 10 16:39:50 GMT 2020 armv7l GNU/Linux
pi@HS3-Pi:~/vb $ mono -V
Mono JIT compiler version 6.8.0.96 (tarball Wed Jan 15 10:23:49 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       normal
        Notifications: epoll
        Architecture:  armel,vfp+hard
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          yes(610)
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```
Test case 1

```
Module Module1

    Class Point
        Public x, y As Double
    End Class

    Sub Main()

        Dim polygon1 As Point() = {
            New Point With {.x = 0.0, .y = 0.0},
            New Point With {.x = 10.0, .y = 0.0},
            New Point With {.x = 10.0, .y = 10.0},
            New Point With {.x = 0.0, .y = 10.0}
        }

        Console.WriteLine("Hello")
    End Sub

End Module

```
Compilation results
```
pi@HS3-Pi:~/vb $ vbnc demo1.vb
Visual Basic.Net Compiler version 0.0.0.5943 (Mono 4.7 - tarball)
Copyright (C) 2004-2010 Rolf Bjarne Kvinge. All rights reserved.

/home/pi/vb/demo1.vb (10,27) : error VBNC90019: Expected '}'.
/home/pi/vb/demo1.vb (10,27): Compiler error around this location, the compiler hasn't implemented the error message, nor error recovery, so the compiler will probably crash soon.
  at vbnc.Helper.ErrorRecoveryNotImplemented (vbnc.Span Location) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseArrayElementInitializer (vbnc.ParsedObject Parent) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseVariableInitializer (vbnc.ParsedObject Parent) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseVariableDeclarator (vbnc.ParsedObject Parent, vbnc.Modifiers Modifiers, vbnc.ParseAttributableInfo Info, System.Collections.IList result, System.Boolean local) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseLocalVariableDeclarator (vbnc.ParsedObject Parent, vbnc.Modifiers Modifiers, vbnc.ParseAttributableInfo Info) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseLocalVariableDeclarators (vbnc.ParsedObject Parent, vbnc.Modifiers Modifiers, vbnc.ParseAttributableInfo Info) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseLocalDeclarationStatement (vbnc.CodeBlock Parent) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseCodeBlock (vbnc.ParsedObject Parent, System.Boolean IsOneLiner) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseSubDeclaration (vbnc.TypeDeclaration Parent, vbnc.ParseAttributableInfo Info) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseTypeMembers (vbnc.TypeDeclaration Parent) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseModuleDeclaration (vbnc.ParsedObject Parent, vbnc.Attributes Attributes, System.String Namespace) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseTypeDeclaration (vbnc.ParsedObject Parent, vbnc.Attributes Attributes, System.String Namespace) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseAssemblyMembers (vbnc.AssemblyDeclaration Parent, System.String RootNamespace) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.ParseAssemblyDeclaration (System.String RootNamespace, vbnc.AssemblyDeclaration assembly) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Parser.Parse (System.String RootNamespace, vbnc.AssemblyDeclaration assembly) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Compiler.Compile_Parse () [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Compiler.Compile () [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Compiler.Compile (System.String[] CommandLine) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Main.Main (System.String[] CmdArgs) [0x00000] in <f876931f1ce142b590f8f6e2605066f5>:0
/home/pi/vb/demo1.vb (10,27) : error VBNC90019: Expected 'End'.
/home/pi/vb/demo1.vb (16,16) : error VBNC90019: Expected 'End'.
/home/pi/vb/demo1.vb (17,12) : error VBNC30203: Identifier expected.
/home/pi/vb/demo1.vb (19,11) : error VBNC30203: Identifier expected.
There were 5 errors and 0 warnings.
Compilation took 00:00:01.9673820

```
I tried to dumb it down with the following

Test case 2
```
Module Module1

    Dim polygon1()() As Double = {
        ({0.0, 0.0}),
        ({10.0, 0.0}),
        ({10.0, 10.0}),
        ({0.0, 10.0})
    }

    Sub Main()
        Console.WriteLine("Hello")
    End Sub

End Module
```
But this fails as well:

```
pi@HS3-Pi:~/vb $ vbnc demo2.vb
Visual Basic.Net Compiler version 0.0.0.5943 (Mono 4.7 - tarball)
Copyright (C) 2004-2010 Rolf Bjarne Kvinge. All rights reserved.

/home/pi/vb/demo2.vb (4,10) : error VBNC99999: Internal compiler error close to this location
vbnc : Command line : error VBNC99999: Unexpected error: Object reference not set to an instance of an object.
  at vbnc.Helper.VerifyValueClassification (vbnc.Expression& Expression, vbnc.ResolveInfo Info) [0x0000e] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.ParenthesizedExpression.ResolveExpressionInternal (vbnc.ResolveInfo Info) [0x0004a] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Expression.ResolveExpression (vbnc.ResolveInfo ResolveInfo) [0x0007b] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Expression.ResolveCode (vbnc.ResolveInfo Info) [0x00001] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.VariableInitializer.ResolveCode (vbnc.ResolveInfo Info) [0x00012] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.BaseList`1[T].ResolveCode (vbnc.ResolveInfo Info) [0x00044] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.ArrayElementInitializer.ResolveCode (vbnc.ResolveInfo Info) [0x000f0] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.VariableInitializer.ResolveCode (vbnc.ResolveInfo Info) [0x00012] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.VariableDeclaration.ResolveCode (vbnc.ResolveInfo Info) [0x001a7] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.BaseObjects`1[T].ResolveCode (vbnc.ResolveInfo Info) [0x00055] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.TypeDeclaration.ResolveCode (vbnc.ResolveInfo Info) [0x00082] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.AssemblyDeclaration.ResolveCode (vbnc.ResolveInfo Info) [0x00041] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Compiler.Compile_Resolve () [0x003bb] in <f876931f1ce142b590f8f6e2605066f5>:0
  at vbnc.Compiler.Compile () [0x004c7] in <f876931f1ce142b590f8f6e2605066f5>:0
Compilation took 00:00:02.7062030
```
If there's a better way to do this, I'd love to hear about it....
Thanks

**Comment by rjvdboon** [comment]
Try using `vbc` instead of `vbnc`. The `vbnc` is a pretty old custom vb compiler implementation, but `vbc` should be the same as the Roslyn based visual basic compiler in Visual Studio.

**Comment by zwolfpack** [comment]
Great - that works. Many thanks!
